lib/services/discourse.ts

```diff
@@ -17,211 +17,80 @@
-	 */
-	public makeGeneric = (data: MessengerEvent): Promise<ReceiptContext> => {
-		// Encode once the common parts of a request
-		const getGeneric = {
-			json: true,
-			method: 'GET',
-			qs: {
-				api_key: this.data.token,
-				api_username: this.data.username,
-			},
-			// appended before execution
-			uri: `https://${this.data.instance}`,
-		};
-		// Gather more complete details of the enqueued event
-		const getPost = _.cloneDeep(getGeneric);
-		getPost.uri += `/posts/${data.rawEvent.id}`;
-		const getTopic = _.cloneDeep(getGeneric);
-		getTopic.uri += `/t/${data.rawEvent.topic_id}`;
-		return Promise.props({
-			post: request(getPost),
-			topic: request(getTopic),
-		})
-		.then((details: {post: any, topic: any}) => {
-			// Gather metadata and resolve
-			const metadata = Messenger.extractMetadata(details.post.raw);
-			const first = details.post.post_number === 1;
-			return {
-				action: MessengerAction.Create,
-				first,
-				genesis: metadata.genesis || data.source,
-				// post_type 4 seems to correspond to whisper
-				hidden: first ? !details.topic.visible : details.post.post_type === 4,
-				source: DiscourseService._serviceName,
-				sourceIds: {
-					// These come in as integers, but should be strings
-					flow: details.topic.category_id.toString(),
-					message: details.post.id.toString(),
-					thread: details.post.topic_id.toString(),
-					url: getTopic.uri,
-					user: details.post.username,
-				},
-				text: metadata.content,
-				title: details.topic.title,
-			};
-		});
+	private connectionDetails: DiscourseConnectionDetails;
+
+	constructor(data: DiscourseConnectionDetails, listen: boolean) {
+		super();
+		// #203: Verify connection data
+		this.connectionDetails = data;
+		if (listen) {
+			this.startListening();
+		}
 	}
 
-	/**
-	 * Promise to turn the generic message format into a specific form to be emitted.
-	 * @param data  Generic message format object to be encoded.
-	 * @returns     Promise that resolves to the emit suitable form.
-	 */
-	public makeSpecific = (data: TransmitContext): Promise<DiscourseEmitContext> => {
-		// Attempt to find the thread ID to know if this is a new topic or not
-		const topicId = data.toIds.thread;
-		if (!topicId) {
-			const title = data.title;
-			if (!title) {
-				throw new Error('Cannot create Discourse Thread without a title');
-			}
-			// A new topic request for discourse
-			return new Promise<DiscourseEmitContext>((resolve) => {
-				resolve({
-					endpoint: {
-						api_key: data.toIds.token,
-						api_username: data.toIds.user,
-					},
-					payload: {
-						category: data.toIds.flow,
-						raw: `${data.text}\n\n---\n${Messenger.stringifyMetadata(data)}`,
-						title,
-						unlist_topic: data.hidden ? 'true' : 'false',
-					}
+	public request(requestOptions: UrlOptions & RequestPromiseOptions): Promise<DiscourseResponse> {
+		// This type massages request-promise into bluebird
+		return new Promise((resolve) => {
```

Comment:
If we're using `request-promise`, why is there a `new Promise` here instead of just returning the result of `request`?